fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/DateTimeExtractAndTransform.java

```diff
@@ -631,19 +636,19 @@ public static Expression strToDate(StringLikeLiteral str, StringLikeLiteral form
                 boolean hasMicroPart = org.apache.doris.analysis.DateLiteral
                         .hasMicroSecondPart(format.getStringValue());
                 return DateTimeV2Literal.fromJavaDateType(DateUtils.getTime(DateUtils.formatBuilder(format.getValue())
-                        .toFormatter(), str.getValue()), hasMicroPart ? 6 : 0);
+                        .toFormatter().withResolverStyle(ResolverStyle.STRICT), str.getValue()), hasMicroPart ? 6 : 0);
```

Comment:
could we refactor `formatBuilder` lightly to avoid call `.toFormatter().withResolverStyle(ResolverStyle.STRICT)` too many times in different place